packages/block-editor/src/components/html-element/index.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * WordPress dependencies
+ */
+import { SelectControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+const htmlElementMessages = {
+	header: __(
+		'The <header> element should represent introductory content, typically a group of introductory or navigational aids.'
+	),
+	main: __(
+		'The <main> element should be used for the primary content of your document only. '
+	),
+	section: __(
+		"The <section> element should represent a standalone portion of the document that can't be better represented by another element."
+	),
+	article: __(
+		'The <article> element should represent a self contained, syndicatable portion of the document.'
+	),
+	aside: __(
+		"The <aside> element should represent a portion of a document whose content is only indirectly related to the document's main content."
+	),
+	footer: __(
+		'The <footer> element should represent a footer for its nearest sectioning element (e.g.: <section>, <article>, <main> etc.).'
+	),
+	nav: __(
+		'The <nav> element should represent a navigation element (e.g. a list of links).'
+	),
+};
+
+/**
+ * Control for selecting the block tagname.
+ *
+ * @param {Object}   props                Component props.
+ * @param {Function} props.onChange       Callback to handle onChange.
+ * @param {Object}   props.tagNameOptions Tagnames to be used in the select control.
+ * @param {string}   props.selectedValue  Selected tag value.
+ *
+ * @return {WPElement}                    Control for selecting the block tagname.
+ */
+export default function HtmlElementControl( {
+	onChange,
+	tagNameOptions,
+	selectedValue,
+} ) {
+	// Default tag name options
+	let options = [
+		{ label: '<div>', value: 'div' },
+		{ label: '<header>', value: 'header' },
+		{ label: '<main>', value: 'main' },
+		{ label: '<section>', value: 'section' },
+		{ label: '<article>', value: 'article' },
+		{ label: '<aside>', value: 'aside' },
+		{ label: '<footer>', value: 'footer' },
+		{ label: '<nav>', value: 'nav' },
+	];
```

Comment:
Suggestion: Maybe we should move this outside of the component, like message values, to avoid initializing default options on every render.

Then we could also use it as the [default parameter value](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters) and get rid of the conditional assignment below.

```js
tagNameOptions = DEFAULT_OPTIONS
```